src/main/java/next/dao/UserDao.java

```diff
@@ -1,63 +1,81 @@
                 user.getName(),
                 user.getEmail());
     }
 
-    public void update(User user) throws SQLException {
-        String sql = "UPDATE USERS SET userId = ?, password = ?, name = ?, email = ? WHERE userId = ?";
-        JdbcTemplate.update(sql,
-                user.getUserId(),
-                user.getPassword(),
-                user.getName(),
-                user.getEmail(),
-                user.getUserId());
+    public void update(User user, String modifyUser) throws DataAccessException {
+        String sql = "UPDATE USERS SET name = ?, email = ? WHERE userId = ?";
+        jdbcTemplate.update(sql,
+                ps -> {
+                    ps.setString(1, user.getName());
+                    ps.setString(2, user.getEmail());
+                    ps.setString(3, modifyUser);
+                });
     }
 
-    public List<User> findAll() throws SQLException {
+    public List<User> findAll() throws DataAccessException {
         String sql = "SELECT userId, password, name, email FROM USERS";
-        ResultSet rs = JdbcTemplate.select(sql, null);
 
-        ArrayList<User> userArrayList = new ArrayList<>();
-        User user = null;
-        if (rs.next()) {
-            user = new User(rs.getString("userId"), rs.getString("password"), rs.getString("name"),
-                    rs.getString("email"));
-            userArrayList.add(user);
-        }
-        return userArrayList;
+        return (List<User>) jdbcTemplate.queryForList(sql, rs -> {
+            ArrayList<User> userArrayList = new ArrayList<>();
+            while (rs.next()){
+                User userCreate = new User(rs.getString("userId"),
+                        rs.getString("password"),
+                        rs.getString("name"),
+                        rs.getString("email"));
+                userArrayList.add(userCreate);
+            }
+            return userArrayList;
+        });
     }
 
-    public User findByUserId(String userId) throws SQLException{
+    public User findByUserId(String userId) throws DataAccessException{
         String sql = "SELECT userId, password, name, email FROM USERS WHERE userId=?";
-        ResultSet rs = JdbcTemplate.select(sql, userId);
+        User paramMap = (User) jdbcTemplate.queryForObject(sql, rs -> {
+            User userCreate = new User(rs.getString("userId"),
+                    rs.getString("password"),
+                    rs.getString("name"),
+                    rs.getString("email"));
+
+            return userCreate;
+        }, userId);
+
+        return paramMap;
+    }
 
-        User user = null;
-        if (rs.next()) {
-            user = new User(rs.getString("userId"), rs.getString("password"), rs.getString("name"),
+    public User findByUserIdSetter(String userId) throws DataAccessException{
+        String sql = "SELECT userId, password, name, email FROM USERS WHERE userId=?";
+        User paramMap = (User) jdbcTemplate.queryForObject(sql, rs -> {
```

Comment:
할당 없이 바로 리턴해주면 어떨까요?

```suggestion
        User paramMap = (User) jdbcTemplate.queryForObject(sql, rs -> new User(
                rs.getString("userId"),
                rs.getString("password"),
                rs.getString("name"),
                rs.getString("email")
             ),
```

아래 람다 사용의 Best Practice 아티클은 참고하시면 앞으로의 람다 표현식 사용에 도움이 될 것 같아요.
특히 8 챕터 (Keep Lambda Expressions Short And Self-explanatory)에 코멘트 드린 내용과 관련 있으니 참고 부탁드려요.

[Best Practices using Java 8 Lambdas | Baeldung](https://www.baeldung.com/java-8-lambda-expressions-tips)